Sources/SwiftInspectorVisitors/Tests/TypeConformanceVisitorSpec.swift

```diff
@@ -0,0 +1,116 @@
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import Nimble
+import Quick
+import Foundation
+import SwiftInspectorTestHelpers
+
+@testable import SwiftInspectorVisitors
+
+final class TypeConformanceVisitorSpec: QuickSpec {
+  private var sut = TypeConformanceVisitor()
+  
+  override func spec() {
+
+    describe("visit(_:)") {
+      context("when a type conforms to a protocol") {
+        context("with only one conformance") {
+          beforeEach {
+            let content = """
+            class SomeObject: Foo {}
+            """
+            self.sut = TypeConformanceVisitor()
+            try? VisitorExecutor.walkVisitor(self.sut, overContent: content)
+          }
+
+          it("returns the conforming type name") {
+            expect(self.sut.conformsToTypes) == ["Foo"]
+          }
+        }
+
+        context("with multiple conformances on the same line") {
+          beforeEach {
+            let content = """
+            class SomeObject: Foo, Bar, FooBar {}
+            """
+            self.sut = TypeConformanceVisitor()
+            try? VisitorExecutor.walkVisitor(self.sut, overContent: content)
+          }
+
+          it("returns the conforming type names") {
+            expect(self.sut.conformsToTypes) == ["Foo", "Bar", "FooBar"]
+          }
+        }
+
+        context("with multiple conformances on multiple lines") {
+          beforeEach {
+            let content = """
+            class SomeObject: Foo, Bar,
+              FooBar {}
+            """
+            self.sut = TypeConformanceVisitor()
+            try? VisitorExecutor.walkVisitor(self.sut, overContent: content)
+          }
+
+          it("returns the conforming type names") {
+            expect(self.sut.conformsToTypes) == ["Foo", "Bar", "FooBar"]
+          }
+        }
+      }
+
+      context("when an inner type conforms to a protocol but the outer type does not") {
+        beforeEach {
+          let content = """
+          class SomeObject {
+            struct SomeStruct: Foo {}
+          }
+          """
+          self.sut = TypeConformanceVisitor()
+          try? VisitorExecutor.walkVisitor(self.sut, overContent: content)
+        }
+
+        it("does not find the inner type's conformance the conforming type names") {
+          expect(self.sut.conformsToTypes).to(beEmpty())
+        }
+      }
+
+      context("when both an outer and inner type conform to a protocol") {
```

Comment:
Isn't this incorrect for this use case?

Reply:
Do you mean the behavior is incorrect? I don't think so. Similar to what I wrote in https://github.com/fdiaz/SwiftInspector/pull/55#discussion_r565742802, I think we want these visitors to be as narrowly scoped as possible